packages/cadl-ranch-specs/http/payload/json-merge-patch/main.tsp

```diff
@@ -0,0 +1,183 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for merge-patch+json content-type")
+@scenarioService("/json-merge-patch")
+namespace Payload.JsonMergePatch;
+
+@doc("Details about a resource.")
+model Resource {
+  name: string;
+  description?: string;
+  map?: Record<InnerModel>;
+  array?: InnerModel[];
+  intValue?: int32;
+  floatValue?: float32;
+  innerModel?: InnerModel;
+  intArray?: int32[];
+}
+
+@doc("Details about a resource for patch operation.")
+model ResourcePatch {
+  description?: string;
+  map?: Record<InnerModel>;
+  array?: InnerModel[];
+  intValue?: int32;
+  floatValue?: float32;
+  innerModel?: InnerModel;
+  intArray?: int32[];
+}
+
+@doc("It is the model used by Resource model")
+model InnerModel {
+  name?: string;
+  description?: string;
+}
+
+@scenario
+@scenarioDoc("""
+
+Expected input body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1,
+  "innerModel": {
+    "name": "InnerMadge",
+    "description": "innerDesc"
+  },
+  "intArray": [1, 2, 3]
+}
+```
+
+Expected response body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
```

Comment:
Can we split this case into (1) a Patch model only contains basic types.   (2) The Patch model with complex types, e.g. array, records, inner models etc. 
The complexity of handling them are quite different.

Reply:
I think the priority regarding to these cases are the same, as there is service team using patch on `Map<InnerModel>` which is a complex case. I listed the case in the description of the pr. 
So I think those are all mandatory..

Reply:
I think we are good having all of these in the same model

Reply:
Thanks @pshao25 . I will resolve the comment, @lirenhe please let me know if you have futrhur concern.